packages/expo-av/ios/EXAV/EXAV.m

```diff
@@ -746,12 +749,36 @@ - (void)sendEventWithName:(NSString *)eventName body:(NSDictionary *)body
 
 #pragma mark - Audio API: Recording
 
+UM_EXPORT_METHOD_AS(getPermissionsAsync,
+                    getPermissionsAsync:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  if (!_permissionsModule) {
+    return reject(@"E_NO_PERMISSIONS", @"Permissions module not found. Are you sure that Expo modules are properly linked?", nil);
+  }
+  [_permissionsModule getPermissionUsingRequesterClass:[EXAudioRecordingRequester class]
+                                            withResult:resolve
+                                          withRejecter:reject];
+}
+
+UM_EXPORT_METHOD_AS(requestPermissionsAsync,
+                    requestPermissionsAsync:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  if (!_permissionsModule) {
+    return reject(@"E_NO_PERMISSIONS", @"Permissions module not found. Are you sure that Expo modules are properly linked?", nil);
+  }
+  [_permissionsModule askForPermissionUsingRequesterClass:[EXAudioRecordingRequester class]
+                                               withResult:resolve
+                                             withRejecter:reject];
+}
+
 UM_EXPORT_METHOD_AS(prepareAudioRecorder,
                     prepareAudioRecorder:(NSDictionary *)options
                     resolver:(UMPromiseResolveBlock)resolve
                     rejecter:(UMPromiseRejectBlock)reject)
 {
-  if (![[_moduleRegistry getModuleImplementingProtocol:@protocol(UMPermissionsInterface)] hasGrantedPermission:@"audioRecording"]) {
+  if (![_permissionsModule hasGrantedPermissionUsingRequesterClass:[EXAudioRecordingRequester class]]) {
```

Comment:
Above you check for `_permissionsModule` existence and here you just invoke some method on it. Please be consistent.
```suggestion
  if (!_permissionsModule) {
    return reject(@"E_NO_PERMISSIONS", @"Permissions module not found. Are you sure that Expo modules are properly linked?", nil);
  }

  if (![_permissionsModule hasGrantedPermissionUsingRequesterClass:[EXAudioRecordingRequester class]]) {
```